server/WAPI.rb

```diff
@@ -252,6 +272,31 @@ def get_request(request)
         wrapped_response = do_request(request)
       end
     end
+
+    # If it didn't work just return that information.
+    logger.debug "#{self.class.to_s}:#{__method__}:#{__LINE__}: wrapped_response: meta_status: #{wrapped_response.meta_status}"
+    if wrapped_response.meta_status != WAPI::SUCCESS
+      return wrapped_response
+    end
+
+    # Ran a query successfully.  See if got partial data and need to keep going.
+    logger.debug "#{self.class.to_s}:#{__method__}:#{__LINE__}: wrapped_response: data length: #{wrapped_response.result.length}"
+
+    # See if there is a link header, if so get the rest of the data.
+    if wrapped_response.meta_more.length > 0
+      logger.debug "#{self.class.to_s}:#{__method__}:#{__LINE__}: found link header: >>#{wrapped_response.meta_more}<<"
+
+      more_data = get_request(wrapped_response.meta_more)
```

Comment:
If the amount of data in the DB could change (grow or expand), then the total page count will change as you request subsequent pages.  So, although recursion seems less safe at first consideration, I think it will be more accurate in the end.